pvanalytics/features/orientation.py

```diff
@@ -0,0 +1,193 @@
+    # spacing given by `freq`.
+    return data.count() * _freqstr_to_hours(freq)
+
+
+def _group_by_day(data):
+    # Group data by timezone localized date.
+    #
+    # We use this function (rather than `data.resample('D')`) because
+    # Resampler.apply() makes the data tz-naive when passed to the
+    # function being applied. This causes the curve fitting functions
+    # to fail when the minute-of-the-day is used as the x-coordinate
+    # since removing timezone information can cause data for a day to
+    # span two dates.
+    #
+    # The date needs to be localized in the same timezone as
+    # data.index so we can easily reindex the series with the original
+    # index at a later time.
+    return data.groupby(
+        pd.to_datetime(data.index.date).tz_localize(data.index.tz)
+    )
+
+
+def tracking_nrel(power_or_irradiance, daytime, correlation_min=0.94,
+                  fixed_max=0.96, min_hours=5, peak_min=None,
+                  midday=None):
+    """Flag days that match the profile of a single-axis tracking PV system.
+
+    For the values on a day to be marked True, they must satisfy the
+    following three conditions:
+
+    1. a restricted quartic [#]_ must fit the data with :math:`r^2`
+       greater than `correlation_min`
+    2. the :math:`r^2` for a quadratic fit must be less than
+       `fixed_max`
+    3. the :math:`r^2` for the restricted quartic fit must be greater
+       than the :math:`r^2` for a quadratic fit
+
+    Values on days where any one of these conditions is not met are
+    marked False.
+
+    .. [#] The specific quartic used for this fit is centered within
+       70 minutes of 12:00, the y-value at the center must be within
+       15% of the median for the day, and it must open downwards.
+
+    Parameters
+    ----------
+    power_or_irradiance : Series
+        Timezone localized series of power or irradiance measurements.
+    daytime : Series
+        Boolean series with True for times that are during the
+        day. For best results this mask should exclude early morning
+        and late afternoon as well as night. These times may have
+        problems with shadows that interfere with curve fitting, but
+        do not necessarily indicate that the tracker is not working.
+    correlation_min : float, default 0.94
+        Minimum :math:`r^2` for a day to be considered sunny.
+    fixed_max : float, default 0.96
+        Maximum :math:`r^2` for a quadratic fit, if the quadratic fit
+        is better than this, then tracking/fixed cannot be determined
+        and the day is marked False.
+    min_hours : float, default 5.0
+        Minimum number of hours with data to attempt a fit on a day.
+    peak_min : float, default None
+        The maximum value for a day must be greater than `peak_min`
+        for a fit to be attempted. If the maximum for a day is less
+        than `peak_min` then the day will be marked False.
+    midday : Series, default None
```

Comment:
Let's combine `daytime` and `midday` into a single mask that selects data to be used in the fitting. The user will need to combine the day/night filter with any time/zenith limits outside this function.

I see your earlier point about two masks being awkward.

Reply:
I'd like to talk this through a bit with @matt14muller. I think that in order to get a reasonably good fit with the particular quartic that he devised it is important to include some of the morning and evening. In my testing, I found that if we were to decrease the maximum zenith angle to 85 degrees the r^2 for the quartic drops to ~0.8 for the tracking system used in the tests. 

Since the point of the `midday` mask is to eliminate the long shadowed periods that are only present when the tracker is not functioning correctly I'm not sure that combining them is the right thing to do. It is a bit awkward, but - at the moment - I think the awkwardness is acceptable since it keeps parameters with different semantics separate. I'm happier with the `None` default than I thought I would be.

Reply:
I think this has been addressed by renaming the parameter and clarifying its purpose in the documentation.